src/machi_cr_client.erl

```diff
@@ -935,3 +923,22 @@ timeout(infinity) ->
     timeout(15*60*1000);                        % close enough to infinity
 timeout(Timeout0) ->
     {Timeout0, Timeout0 + 30*1000}.
+
+trim_both_side([], _Offset, _Size) -> [];
+trim_both_side([{F, Offset0, Chunk, _Csum}|L], Offset, Size)
+  when Offset0 < Offset ->
+    TrashLen = 8 * (Offset - Offset0),
+    <<_:TrashLen/binary, NewChunk/binary>> = Chunk,
+    NewH = {F, Offset, NewChunk, <<>>},
+    trim_both_side([NewH|L], Offset, Size);
+trim_both_side(Chunks, Offset, Size) ->
+    %% TODO: optimize
+    [{F, Offset1, Chunk1, _Csum1}|L] = lists:reverse(Chunks),
```

Comment:
If I understand this correctly, you're trying to get the last element of the list `Chunks` - could you use lists:last/1 instead of reversing the whole list?